test/test_helper.rb

```diff
@@ -89,5 +98,34 @@ def after_node(node)
         # Ignore, too noisy
       end
     end
+
+    # This check flags uses of the assign tag.
+    class AssignCheck < ThemeCheck::LiquidCheck
+      def on_assign(node)
+        add_offense("assign used", node: node)
+      end
+    end
+
+    # This check flags /RegexError \d+/ in the document
+    class RegexCheck < ThemeCheck::LiquidCheck
+      include ThemeCheck::RegexHelpers
+
+      def initialize
+        @regex = /RegexError \d+/
+        @offenses = []
+      end
+
+      def on_document(node)
+        source = node.template.source
+        matches(source, @regex).each do |match|
+          add_offense(
+            "Regex found #{match[0]}.",
+            node: node,
+            markup: match[0],
+            line_number: source[0...match.begin(0)].count("\n") + 1
+          )
+        end
+      end
+    end
```

Comment:
Is there really value in adding these here, they are only used in `DisabledChecksTest`?

And these bleed into `DocumentationTest`.

If you nest them under `DisabledChecksTest`, they will be isolated to that test.

Reply:
Actually, I think you are right. The tests initially were using TracerCheck which _is_ reused in other places. So I put those there as I was changing things. I like that idea. I'll do that.

Reply:
Actually it's because they depend on LiquidCheck (which does the CheckTracking). I did the offense << Offense.new manually and extend Check directly with no problems.